src/Hyperbee.Json/JsonPath.cs

```diff
@@ -30,48 +30,20 @@
 //
 #endregion
 
-using System.Collections.Immutable;
-using System.Globalization;
 using System.Text.Json;
-using System.Text.RegularExpressions;
 using Hyperbee.Json.Evaluators;
-using Hyperbee.Json.Extensions;
-using Hyperbee.Json.Memory;
 using Hyperbee.Json.Tokenizer;
 
 namespace Hyperbee.Json;
 // https://ietf-wg-jsonpath.github.io/draft-ietf-jsonpath-base/draft-ietf-jsonpath-base.html
 // https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base
 
-public sealed partial class JsonPath
+public sealed class JsonPath
 {
     public static IJsonPathScriptEvaluator<JsonElement> DefaultEvaluator { get; set; } = new JsonPathCSharpElementEvaluator();
     private readonly IJsonPathScriptEvaluator<JsonElement> _evaluator;
 
-    // generated regex
-
-    [GeneratedRegex( "^(-?[0-9]*):?(-?[0-9]*):?(-?[0-9]*)$" )]
-    private static partial Regex RegexSlice();
-
-    [GeneratedRegex( @"^\?\((.*?)\)$" )]
-    private static partial Regex RegexPathFilter();
-
-    [GeneratedRegex( @"^[0-9*]+$" )]
-    private static partial Regex RegexNumber();
-
-    // property names can be simple (@.property) if they contain no SpecialCharacters,
-    // otherwise they require bracket notation (@['property']).
-
-    private static readonly char[] SpecialCharacters = ['.', ' ', '\'', '/', '"', '[', ']', '(', ')', '\t', '\n', '\r', '\f', '\b', '\\', '\u0085', '\u2028', '\u2029'];
-    private static string GetPath( string prefix, int childKey ) => $"{prefix}[{childKey}]";
-
-    private static string GetPath( string prefix, string childKey, JsonValueKind tokenKind )
-    {
-        if ( tokenKind == JsonValueKind.Array )
-            return $"{prefix}[{childKey}]";
-
-        return childKey.IndexOfAny( SpecialCharacters ) == -1 ? $"{prefix}.{childKey}" : $@"{prefix}['{childKey}']";
-    }
+    private readonly JsonDocumentPathVisitor _visitor = new();
```

Comment:
I moved all the JSONDocument and JSONElement-specific code into a visitor that only gets newed up once.